docs/contracts-intro.md

```diff
@@ -2,11 +2,159 @@
+
+## Common and Custom Contracts
+
+Data backends that are accessible via OpenSAFELY provide data tables. Each
+data table satisfies an OpenSAFELY Contract.
+
+There are two types of Contracts:
+
+* **Common Contracts**, required by all OpenSAFELY backends and covering
+  core clinical data domains
+* **Custom Contracts**, offered by just some backends and covering more
+  specialised clinical data domains
+
+Both the structure of the Contracts and the method of accessing
+associated tables is the same for both Common and Core Contracts.
+
+### Common Contracts
+
+All OpenSAFELY backends satisfy Common Contracts.
+
+There will be several different Common Contracts available with the
+initial release of Data Builder. More Common Contracts may be added in
+future.
+
+!!! warning
+    Some backends might deviate slightly from the suggested OpenSAFELY
+    behaviour shared by most backends. Refer to the [OpenSAFELY
+    Contracts Reference](contracts-reference.md) for backends you intend
+    to access with your dataset definition.
+
+For dataset definitions that only access tables defined by Common
+Contracts, those dataset definitions should be largely compatible with all
+OpenSAFELY backends.
+
+### Custom Contracts
+
+Some backends may offer particularly specialised data that is not yet
+covered by a Common Contract. *Custom Contracts* allow individual
+backends to still offer this data via OpenSAFELY.
+
+!!! warning
+    Using Custom Contracts might tie a dataset definition more closely
+    to a specific backend, than if just Common Contracts are used. This
+    also depends on how the dataset definition is designed.
+
+!!! note
+    Should additional data providers want to offer data that closely resembles
+    an existing *Custom* Contract, OpenSAFELY developers would consider a new
+    *Common* Contract.
+
+## Reading an OpenSAFELY Contract
+
+!!! note
+    Refer to the [OpenSAFELY Contracts
+    Reference](contracts-intro.md#the-opensafely-contracts-reference) for
+    the details of specific contracts.
+
+Each OpenSAFELY Contract relates to a specific domain of health record
+data. The name of the Contract indicates the data domain covered by the
+Contract. Tables that satisfy these Contracts can be accessed in a
+dataset definition via [Data Builder's query language,
+ehrQL](ehrql-intro.md).
+
+An OpenSAFELY Contract provides the following information for each
+associated table:
+
+* the **column names**. Each column is denoted by the identifier you use
+  to access it in ehrQL.
+* the **column's description**. The description summarises what the
+  column represents.
+* the **column's data type**. For instance, a column's data type might
+  be one of the following:
+    * integer type
+    * floating point type
+    * date type
+    * string type
+    * or some other type not in this list
+* the **additional constraints on the column's data values**, other than
```

Comment:
We should mention that our plan is for these to be validated for researchers, so they can expect these conditions to always be true, but at the moment they're informational only.